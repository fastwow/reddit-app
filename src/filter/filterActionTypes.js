export const FILTER_POSTS = 'filter_posts';
