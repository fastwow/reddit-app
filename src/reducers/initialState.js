export default {
  posts: {isLoading: false, isRefreshing: false, errorMessage: '', posts: [], after: ''},
  favorites: {posts: []},
  filter: {searchTerm: ''},
};
