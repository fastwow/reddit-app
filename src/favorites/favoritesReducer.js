import initialState from './favoritesActionsInitialState';
import {ADD_TO_FAVORITE, REMOVE_FROM_FAVORITE} from './favoritesActionTypes';

export default function (state = initialState, action) {
  switch (action.type) {
    case ADD_TO_FAVORITE:
      return {
        ...state,
        posts: state.posts.concat(action.data),
      };
    case REMOVE_FROM_FAVORITE:
      return {
        ...state,
        posts: state.posts.filter(item => action.data.id !== item.id),
      };
    default:
      return state;
  }
}
