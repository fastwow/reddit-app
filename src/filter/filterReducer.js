import {FILTER_POSTS} from './filterActionTypes';
import initialState from './filterActionsInitialState';

export default function (state = initialState, action) {
  switch (action.type) {
    case FILTER_POSTS:
      return {
        ...state,
        searchTerm: action.searchTerm,
      };
    default:
      return state;
  }
}
