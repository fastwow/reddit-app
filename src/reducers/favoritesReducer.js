import initialState from './initialState';
import {ADD_TO_FAVORITE, REMOVE_FROM_FAVORITE} from '../actions/types';

export default function (state = initialState.favorites, action) {
  switch (action.type) {
    case ADD_TO_FAVORITE:
      return {
        ...state,
        posts: state.posts.concat(action.data),
      };
    case REMOVE_FROM_FAVORITE:
      return {
        ...state,
        posts: state.posts.filter(item => action.data.id !== item.id),
      };
    default:
      return state;
  }
}
