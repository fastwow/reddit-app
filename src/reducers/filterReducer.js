import {
  FILTER_POSTS,
} from '../actions/types';
import initialState from './initialState';

export default function (state = initialState.filter, action) {
  switch (action.type) {
    case FILTER_POSTS:
      return {
        ...state,
        searchTerm: action.searchTerm,
      };
    default:
      return state;
  }
}
