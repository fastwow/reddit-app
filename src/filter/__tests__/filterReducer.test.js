import initialState from '../filterActionsInitialState';
import filterReducer from '../filterReducer';
import {FILTER_POSTS} from '../filterActionTypes';
import favoritesReducer from '../../favorites/favoritesReducer';

describe('filterReducer', () => {
  it('returns the same state for filter reducer on an unhandled action', () => {
    expect(filterReducer(initialState, {type: '_NULL'})).toMatchSnapshot();
  });

  it(`handles ${FILTER_POSTS} action`, () => {
    expect(favoritesReducer(initialState, {searchTerm: 'cat', type: FILTER_POSTS})).toMatchSnapshot();
  });
});

