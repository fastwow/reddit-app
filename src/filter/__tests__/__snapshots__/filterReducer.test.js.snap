// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterReducer handles filter_posts action 1`] = `
Object {
  "searchTerm": "",
}
`;

exports[`filterReducer returns the same state for filter reducer on an unhandled action 1`] = `
Object {
  "searchTerm": "",
}
`;
